tsl/test/expected/bgw_job_ddl.out

```diff
@@ -39,10 +44,29 @@ ERROR:  permission denied to reassign objects
 RESET ROLE;
 -- Test that renaming a user changes keeps the job assigned to that user.
 ALTER USER another_user RENAME TO renamed_user;
-SELECT proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
-   proc_name    |    owner     
-----------------+--------------
- bgw_test_job_1 | renamed_user
+SELECT proc_schema, proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
+ proc_schema | proc_name  |    owner     
+-------------+------------+--------------
+ public      | more_magic | renamed_user
+(1 row)
+
+-- Test that renaming the procedure also modifies the entry in the
+-- jobs table.
+ALTER PROCEDURE more_magic RENAME TO magic;
+SELECT proc_schema, proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
+ proc_schema | proc_name |    owner     
+-------------+-----------+--------------
+ public      | magic     | renamed_user
+(1 row)
+
+-- Test that modifying the schema also modifies the entry in the jobs
+-- table.
+CREATE SCHEMA frugal;
+ALTER PROCEDURE magic SET SCHEMA frugal;
```

Comment:
What about renaming the schema? Or dropping the schema or the procedure?

Reply:
Added code to deal with renaming the schema so that it will rename the schema name of the job procedure as well.

Added tests to check that dropping the schema cannot be done.

Added code to deal with dropping a function used by a background job.